### PR TITLE
WebHost: Refactor `tracker.py`, removal of dead code, and tweaks to layouts of some tracker pages.

### PR DESCRIPTION
**Pull request by ThePhar**
## What is this fixing or adding?
This is a huge refactor of `tracker.py` in WebHost and depending files.

Tracker API Changes:
- Removes all the root-level LTTP functions/variables and moves them to LTTP render functions.
- Remove imports of `item_id_to_name`/`location_id_to_name` lookup tables.
  - These are also completely removed from `worlds/__init__.py` as they are no longer needed.
- Creates a "preliminary" Tracker API helper object to build potential objects/lookups for each world that wants to implement their own tracker.
  - Long term goal is to finalize what functions are available, create documentation, and move world-specific render functions, templates, and resources to their cooresponding world once future of WebHost is decided. For now, this is a middle ground.
  - Imports `network_data_package` only to check if game exists in datapackage to prevent WebHost from loading the tracker if relevant world folder/apworld is removed on same machine WebHost is running from.
- Item/Location lookup dicts/functions check the game to prepare for #1933 which will allow games to overlap ids so knowing the game is important.
- Subsequent calls on TrackerData methods will return cached results for that request to prevent developers from computing more than necessary.

Multiworld/Player Template Changes:
- All player-specific custom game trackers now temporarily include a link to the generic tracker for better UX.
- All player-specific generic trackers now include a link to any custom game tracker, if one exists.
- All player-specific generic trackers now include relevant hints.
  - Relevant player's name is bolded on this hints table.
- All tracker tables can now be resized by the user.
- Added a link to Multiworld Tracker from player-specific generic trackers.
- Improved "Multistream" link when visible on Multiworld Tracker page.
- Added "game" column to hints columns to show which game the item is contained in.
- Tweaked Factorio multiworld tracker to match unicode symbols as ALTTP multiworld tracker.
- Tweaked ALTTP multiworld tracker to display number of obtained Triforce Pieces.
- Added hover text for icons on Factorio and ALTTP multiworld trackers.
- Only show relevant players on game-specific multiworld trackers instead of all players.
  - Also hides game column, since it's inferred.
- Slight style changes to navigation/search/info text.

CustomServer Changes:
- Defined YouTube and Twitch as only supported platforms in `!video` command doccomment.

MISC Changes:
- Code style improvements.
- Rename of some files/functions.
- Removed `lookup_any_item_id_to_name` and `lookup_any_location_id_to_name` functions.
- Removed logging for "custom mode" worlds that use `data_version`: 0 as it doesn't current do anything in AP core anyway.
  - Only relevant for legacy clients that do not support `checksum`s, which will be dropped in #1933, anyway.

## How was this tested?
Thoughoughly generated multiple rooms with a variety of games that both include and do not include custom trackers.

- Verified all multiworld trackers and game trackers still worked correctly and updated when items were received, locations were checked, and hints were broadcasted.
- Verified multistream link renders properly and links correctly.
- Verified links to and from generic trackers work.

## If this makes graphical changes, please attach screenshots.
image
image
image
image
image
image
image
image

### CODE REVIEW

**Comment by ThePhar** [comment]
Requested reviews from all relevant world owners (and @jonloveslegos as you haven't accepted collaborator invite) since I touched your world's tracker code.

Also Farrak for the WebHost front-end changes.

**Comment by ThePhar** [comment]
Some form of these changes will be required to complete #1933.

**Review by Ziktofel — APPROVED** [review]
Looks good for SC2 as long as it works

File: WebHostLib/static/styles/tracker.css [hunk 1]
```diff
@@ -7,149 +7,160 @@
...
-}
-
-#tracker-header-bar{
+#tracker-header-bar {
     display: flex;
     flex-direction: row;
     justify-content: flex-start;
+    align-content: center;
     line-height: 20px;
+    gap: 0.5rem;
```
Comment (LegendaryLinux) [inline WebHostLib/static/styles/tracker.css:22]: RIP Internet Explorer users 🤣.

**Review by LegendaryLinux — APPROVED** [review]
I do have a minor UX flow issue.

The generic multiworld tracker has this UI:
image

Clinking on the slot for LttP gives you this:
image

And clicking on "Switch to Generic Tracker" gives you this:
image

Which returns to the multiworld tracker, which is labelled "Generic."

Not a huge issue, and I think people will figure out the flow easily enough, but I guarantee some will be confused.

Other than that, HTML / CSS / JS looks good to me.

File: WebHostLib/static/styles/tracker.css (hunk 1 above)
Comment (ThePhar) [reply WebHostLib/static/styles/tracker.css:22]: Good.

**Comment by lordlou** [comment]
I didnt do the SM tracker. I think its @TauAkiou.

**Pull request merged**